entity_extractor.py

```diff
@@ -0,0 +1,72 @@
+"""NER Entity Token Classifier"""
+from typing import Dict, List, Any
+from pydantic import Field
+from transformers import (
+  pipeline,
+  AutoTokenizer,
+  AutoModelForTokenClassification)
+
+class EntityExtractor():
+  """Entity Extractor"""
+
+  _input_text: str = Field(default=None)
+  _predictions: List[Dict[str, Any]] = Field(default=[])
+  _locations: List[str] = Field(default=[])
+  _organsiations: List[str] = Field(default=[])
+  _persons: List[str] = Field(default=[])
+
+  def __init__(
+    self,
+    model_name:str="Jean-Baptiste/roberta-large-ner-english"):
+
+
+    self._model_name = model_name
+
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+
+    model = AutoModelForTokenClassification.from_pretrained(model_name)
+
+    self._model = pipeline('ner', model=model, tokenizer=tokenizer, aggregation_strategy="simple")
+
+  def inference(self, input_text:str) -> Dict:
+    """inference function"""
+
+    self._input_text = input_text
+
+    self._predictions = self._model(self._input_text)
+
+    return self._predictions
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `inference` method, you're assigning the input text to `_input_text` and then using it immediately in the next line. It would be more efficient to use the `input_text` parameter directly. Also, consider adding error handling for cases where the inference fails.

```diff
-    self._input_text = input_text
-    self._predictions = self._model(self._input_text)
+    try:
+        self._predictions = self._model(input_text)
+    except Exception as e:
+        raise RuntimeError(f"Inference failed on input text. Error: {str(e)}")
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->